### PR TITLE
Create Initial Documentation Guide

### LINKED ISSUE #41: Where Should Engineering Documentation Go?
Branching off of issue #40, a question embedded in that thread is: where is the canonical location for engineering documentation? 

I'll respond here with some options that you 👍 or 👎 and @desiringgod/engineers you can propose other ideas.

### PR DESCRIPTION
**Pull request by hchasey**
Closes #41

While there could be a lot more details, this is attempting to take the first step and close one question.

### CODE REVIEW

**Comment by chrismaddocks** [comment]
@hchasey What do you think about mentioning the wiki here? I'm not recommending we put stuff there. Since it _does_ exist, though, maybe we acknowledge its existence, mention what kinds of things are there, and clarify if we want that to be true going forward?

**Comment by chrismaddocks** [comment]
Or, if the `www` wiki exists because it's repo-specific, call that out too. It seems like a large (if aging) part of our tech documentation lives there at the moment.

**Comment by hchasey** [comment]
Yeah @chrismaddocks – that is a good idea and I considered mentioning where and how to document within a specific non-guides repo, but I wanted to propose that in a separate PR since there could be some discussion surrounding wiki vs `/docs` etc.

File: workflow/documentation/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+## Where to Document
+
+By default, start with [this desiringgod/guides](https://github.com/desiringgod/guides) repository and much of what you need to document will fit here.
+
+If what you are documenting contains confidential information, it should be documented elsewhere, and you can ask the team or open an issue or pull request in a related private repository.
+
+### Documenting in Other Repos
+
+If what you are documenting is not related to a [Product Engineering Team](https://github.com/orgs/desiringgod/teams/engineers/members) guide, workflow, standard, or style, it should likely be documented elsewhere, typically within another repository within the [desiringGod organization](https://github.com/desiringgod).
```
Comment (joeosburn-desiringgod) [inline workflow/documentation/README.md]: Are we concerned about the fact that this link: https://github.com/orgs/desiringgod/teams/engineers/members is private?

Comment (hchasey) [inline workflow/documentation/README.md]: Good question @joeosburn-desiringgod – everyone that is in the org can see it, so I think I am ok with it for now, but am happy to hear other view points.

**Comment by hchasey** [comment]
@benhutton – I don't disagree with you. I think that what and how will be helpful additions to this documentation guide and/or [the guides home page](https://github.com/desiringgod/guides/blob/master/README.md). However, I think I'd like this to be a separate PR/discussion from this initial version of the guide. Then, we can get into the details there. What do you think of getting this initial version in and then you and/or other can propose what and how and FAQs additions? 

Then, @joeosburn-desiringgod, @pjmelling, @isaachvazquez – check this out and approve if this looks good enough to move forward with.

File: workflow/documentation/README.md (hunk 1 above)
Comment (joeosburn-desiringgod) [inline workflow/documentation/README.md]: @hchasey It feels odd to me to have a public repo with internal private links. This particular instance is probably not a big deal, but it does highlight the tension point this creates. And, the reality that we could have more instances like this in the future. All in all, I'd still rather we just keep this repo private as I don't see any benefit to DG in keeping this public.

**Comment by hchasey** [comment]
@isaachvazquez @pjmelling – this should be a pretty quick review. I'm waiting on 👍 from you before merging.

**Comment by pjmelling** [comment]
@hchasey where are we at on the question of public vs private access on this repo?

**Comment by hchasey** [comment]
@pjmelling – let's start the public/private repo question in a different thread or issue.

This doc does include:

> If what you are documenting contains confidential information, it should be documented elsewhere....

**Pull request merged**